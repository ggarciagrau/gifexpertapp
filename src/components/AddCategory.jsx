import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onChange = ({ target }) => {
        setInputValue(target.value);
    }

    const onSubmit = (e) => {
        e.preventDefault();
        
        let cleanInputValue = inputValue.trim();
        if (!cleanInputValue.length) return;
        onNewCategory(cleanInputValue);
        setInputValue('');
        // setCategories(categories => [inputValue, ...categories]);
    }

    return (
        <div>
            <form onSubmit={onSubmit} aria-label="form">
                <input
                    type="text"
                    placeholder="Search gifs..."
                    value={inputValue}
                    onChange={onChange}
                />
            </form>
        </div>
    );
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}