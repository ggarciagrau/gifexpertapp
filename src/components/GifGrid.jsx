import PropTypes from "prop-types";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            { isLoading && <h2>Cargando...</h2> }
            <div className="card-grid">
                {
                    images.map(({ id, title, url }) => (
                        <GifGridItem 
                            key={id}
                            {...{title, url, id}}
                        />
                    ))
                }
            </div>
        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}