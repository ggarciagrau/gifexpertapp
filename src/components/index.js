export * from './AddCategory';
export * from './GifGrid';
export *from './GifGridItem';

// import { AddCategory } from "./AddCategory";
// import { GifGrid } from "./GifGrid";
// import { GifGridItem } from "./GifGridItem";

// export {
//     AddCategory,
//     GifGrid,
//     GifGridItem
// };