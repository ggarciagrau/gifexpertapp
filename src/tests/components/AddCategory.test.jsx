import { render, fireEvent, screen } from "@testing-library/react";
import { AddCategory } from "../../components";

describe("AddCategory tests", () => {

    const onNewCategory = () => { };

    test("Renders", () => {

        const { container } = render(<AddCategory onNewCategory={onNewCategory} />);

        expect(container).toMatchSnapshot();
    });

    test("Must change the input value", () => {

        const value = "Hello World";
        const { container } = render(<AddCategory onNewCategory={onNewCategory}/>);

        // const input = screen.getByRole('textbox');
        const input = container.querySelector('input');

        fireEvent.change(input, { target: { value } });

        expect(input.value).toBe(value);
    });

    test("Must call onNewCategory if the input has a value", () => {
        const value = "Hello World";
        const { container } = render(<AddCategory onNewCategory={onNewCategory}/>);

        const form = screen.getByRole('form');
        // const form = container.querySelector('form');
        const input = container.querySelector('input');

        fireEvent.change(input, { target: { value } });
        fireEvent.submit(form);

        expect(input.value).toBe("");
    })
});