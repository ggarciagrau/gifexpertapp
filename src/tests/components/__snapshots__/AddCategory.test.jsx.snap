// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory tests Renders 1`] = `
<div>
  <div>
    <form
      aria-label="form"
    >
      <input
        placeholder="Search gifs..."
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;
